### PR TITLE
etcdctl: support watch with range_end

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/5575.

### CODE REVIEW

**Comment by gyuho** [comment]
/cc @heyitsanthony @xiang90 

Thanks!

File: etcdctl/ctlv3/command/watch_command.go [hunk 1]
```diff
@@ -52,17 +52,21 @@ func watchCommandFunc(cmd *cobra.Command, args []string) {
...
-	if len(args) != 1 {
-		ExitWithError(ExitBadArgs, fmt.Errorf("watch in non-interactive mode requires an argument as key or prefix"))
+	if len(args) < 1 || len(args) > 2 {
+		ExitWithError(ExitBadArgs, fmt.Errorf("watch in non-interactive mode requires one or two arguments as key or prefix, with range end"))
 	}
 
 	opts := []clientv3.OpOption{clientv3.WithRev(watchRev)}
+	key := args[0]
+	if len(args) == 2 {
+		opts = append(opts, clientv3.WithRange(args[1]))
```
Comment (heyitsanthony) [inline etcdctl/ctlv3/command/watch_command.go]: error out if watchPrefix is set?

File: etcdctl/ctlv3/command/watch_command.go [hunk 2]
```diff
@@ -111,6 +115,9 @@ func watchInteractiveFunc(cmd *cobra.Command, args []string) {
 			key = moreargs[0]
 		}
 		opts := []clientv3.OpOption{clientv3.WithRev(watchRev)}
+		if len(moreargs) == 2 {
+			opts = append(opts, clientv3.WithRange(moreargs[1]))
```
Comment (heyitsanthony) [inline etcdctl/ctlv3/command/watch_command.go]: same here

**Comment by gyuho** [comment]
@heyitsanthony PTAL. Just added `prefix` check.

Thanks.

File: etcdctl/ctlv3/command/watch_command.go [hunk 3]
```diff
@@ -111,6 +118,12 @@ func watchInteractiveFunc(cmd *cobra.Command, args []string) {
 			key = moreargs[0]
 		}
 		opts := []clientv3.OpOption{clientv3.WithRev(watchRev)}
+		if len(moreargs) == 2 {
+			if watchPrefix {
+				fmt.Fprintf(os.Stderr, "`range_end` and `--prefix` cannot be set at the same time, choose one\n")
```
Comment (heyitsanthony) [inline etcdctl/ctlv3/command/watch_command.go]: this should probably `continue` like the `len(moreargs)` check

Comment (gyuho) [inline etcdctl/ctlv3/command/watch_command.go]: You are right. Just added. Thanks.

**Comment by heyitsanthony** [comment]
lgtm

**Comment by gyuho** [comment]
thanks. merging in greens.

**Pull request merged**